processor/src/main/java/org/mapstruct/ap/internal/model/source/BeanMappingOptions.java

```diff
@@ -31,7 +31,7 @@
 public class BeanMappingOptions extends DelegatingOptions {
 
     private final SelectionParameters selectionParameters;
-    private final Optional<BeanMappingGem> beanMapping;
```

Comment:
Ok.. At the time of writing it felt more or less ok to do this (we had a discussion on whether or not to pass optional as parameter). Now I found some more post and I agree. `Optional` should not be a member. Our 'chain of responsibility' ensures the result of most methods is actually not-null, so there's also no need to return it. Remains, its fluent use.

Reply:
Sorry, I really don't like using `Optional` as class member or passed as parameters. Thanks for understanding

Reply:
but, that's what I meant.. You're right..

Reply:
I know, just wanted to comment on your comment :)